src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
Hm, every exception is caught here: https://github.com/php-http/promise/blob/master/src/RejectedPromise.php#L36

@joelwurtz would your PR to HTTPlug solve this issue?